app/views/govuk_publishing_components/components/_title.html.erb

```diff
@@ -21,13 +22,20 @@
   heading_classes = %w[gem-c-title__text]
   heading_classes << (average_title_length.present? ? 'govuk-heading-l' : 'govuk-heading-xl')
 %>
+
 <%= content_tag(:div, class: classes) do %>
-  <% if context %>
+  <% if context && !context_inside %>
     <span class="govuk-caption-xl gem-c-title__context" <%= "lang=#{context_locale}" if context_locale.present? %>>
       <%= context_href ? link_to(context_text, context_href, class: 'gem-c-title__context-link govuk-link', data: context_data) : context_text %>
     </span>
   <% end %>
+  
   <h1 class="<%= heading_classes.join(" ") %>">
+    <% if context && context_inside %>
+      <span class="govuk-caption-xl gem-c-title__context" <%= "lang=#{context_locale}" if context_locale.present? %>>
+        <%= context_href ? link_to(context_text, context_href, class: 'gem-c-title__context-link govuk-link', data: context_data) : context_text %>
```

Comment:
It looks like these lines are the same as lines 28-30? In which case you can use the rails `capture` method, something like this:

```
<% context_block = capture do %>
    <span class="govuk-caption-xl gem-c-title__context" <%= "lang=#{context_locale}" if context_locale.present? %>>
      <%= context_href ? link_to(context_text, context_href, class: 'gem-c-title__context-link govuk-link', data: context_data) : context_text %>
    </span>
<% end %>

...

<% if context && !context_inside %>
  <%= context_block %> 
<% end %>

<h1>
  <% if context &&  context_inside %>
    <%= context_block %>
  <% end %>
</h1>

```